x-pack/legacy/plugins/apm/server/lib/service_map/setup_required_scripts.ts

```
@@ -0,0 +1,298 @@
+                ctx['destination.address']=spanName;
+              }
+
+              if (ctx['span']['type'] == 'resource') {
+                def spanName = ctx['span']['name'];
+
+                if (spanName.indexOf('://') > -1) {
+                  spanName = spanName.substring(spanName.indexOf('://')+3, spanName.length());
+                }
+                if (spanName.indexOf('/') > -1) {
+                  spanName = spanName.substring(0, spanName.indexOf('/'));
+                }
+
+                ctx['destination.address']=spanName;
+              }
+
+              if (ctx['span']['type'] == 'db') {
+                def dest = ctx['span']['subtype'];
+                ctx['destination.address']=dest;
+              }
+
+              if (ctx['span']['type'] == 'cache') {
+                def dest = ctx['span']['subtype'];
+                ctx['destination.address']=dest;
+              }
+            }
+          `
+        }
+      ]
+    }
+  });
+}
+
+interface ApmIngestPipeline {
+  [APM_INGEST_PIPELINE_ID]: {
+    description: string;
+    processors: Array<{
+      pipeline: {
+        name: string;
+      };
+    }>;
+  };
+}
+
+async function getIngestPipelineApm(
+  callCluster: CallCluster
+): Promise<ApmIngestPipeline> {
+  return await callCluster('ingest.getPipeline', {
+    id: APM_INGEST_PIPELINE_ID
+  });
+}
+
+async function putIngestPipelineApm(
+  callCluster: CallCluster,
+  processors: ApmIngestPipeline[typeof APM_INGEST_PIPELINE_ID]['processors']
+) {
+  return await callCluster('ingest.putPipeline', {
+    id: APM_INGEST_PIPELINE_ID,
+    body: {
+      description: 'Default enrichment for APM events',
+      processors
+    }
+  });
+}
+
+async function applyExtractDestinationToApm(callCluster: CallCluster) {
+  let apmIngestPipeline: ApmIngestPipeline;
+  try {
+    // get current apm ingest pipeline
+    apmIngestPipeline = await getIngestPipelineApm(callCluster);
+  } catch (error) {
+    if (error.statusCode !== 404) {
```

Maybe add a comment about why we are swallowing 404 errors

---

nvm. comment below is sufficient 👍